internal/rest/getv2clustersname.go

```diff
@@ -105,14 +106,25 @@ func (s *Server) getCluster(ctx context.Context, activeProjectID, name string) (
 		slog.Error("failed to get nodes", "cluster", capiCluster.Name, "error", err)
 		return api.ClusterDetailInfo{}, fmt.Errorf("failed to get nodes, err: %w", err)
 	}
+	if clusterDetailInfo.Nodes == nil || len(*clusterDetailInfo.Nodes) == 0 {
```

Comment:
this condition will always be true

shouldn't you check nodes instead? (gathered in line 104)
```suggestion
	if nodes == nil || len(nodes) == 0 {
```

Reply:
updated in b579c9976dc82390b6c4055c5dabb1e731143be8